### PR TITLE
OpenSlides ♥ Trees

### PR DESCRIPTION
**Pull request by FinnStutzenstein**
We should wait for #4018 to be merged to include the TreeService there

### CODE REVIEW

File: client/src/app/site/base/base-repository.ts [hunk 1]
```diff
@@ -134,7 +138,7 @@ export abstract class BaseRepository<V extends BaseViewModel, M extends BaseMode
      * return the Observable of the whole store
      */
     public getViewModelListObservable(): Observable<V[]> {
-        return this.viewModelListSubject.asObservable();
+        return this.viewModelListSubject.asObservable().pipe(auditTime(1));
```
Comment (tsiegleauq) [inline client/src/app/site/base/base-repository.ts]: requires documentation (esp. the audit time reasoning)

File: client/src/app/site/motions/components/call-list/call-list.component.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export class CallListComponent extends BaseViewComponent {
      * Handler for the sort event. The data to change is given to
      * the repo, sending it to the server.
      */
-    public sort(data: OSTreeSortEvent): void {
+    public sort(data: OSTreeSortEvent<ViewMotion>): void {
```
Comment (tsiegleauq) [inline client/src/app/site/motions/components/call-list/call-list.component.ts]: missing parameter documentation

File: client/src/app/site/motions/services/motion-repository.service.ts [hunk 3]
```diff
@@ -77,6 +82,20 @@ export class MotionRepositoryService extends BaseRepository<ViewMotion, Motion>
         return new ViewMotion(motion, category, submitters, supporters, workflow, state, item);
     }
 
+    public getViewModelListObservable(): Observable<ViewMotion[]> {
+        return this.viewModelListSubject.asObservable().pipe(
+            auditTime(1),
```
Comment (tsiegleauq) [inline client/src/app/site/motions/services/motion-repository.service.ts]: audit time was in base repository. Was this still required here?

**Comment by FinnStutzenstein** [comment]
@tsiegleauq All done. Also did a better checking when dropping things in the tree. Your described case should be prevented now.

File: client/src/app/site/base/base-repository.ts (hunk 1 above)
Comment (FinnStutzenstein) [reply client/src/app/site/base/base-repository.ts]: Done above.

**Pull request merged**